src/cargo/util/toml/targets.rs

```diff
@@ -603,13 +616,17 @@ fn normalize_targets_with_legacy_path(
     autodiscover_flag_name: &str,
 ) -> CargoResult<Vec<TomlTarget>> {
     if are_normalized(toml_targets, autodiscover) {
-        let toml_targets = toml_targets.cloned().unwrap_or_default();
-        for target in &toml_targets {
+        let mut toml_targets = toml_targets.cloned().unwrap_or_default();
+        for target in toml_targets.iter_mut() {
             // Check early to improve error messages
             validate_target_name(target, target_kind_human, target_kind, warnings)?;
 
             validate_proc_macro(target, target_kind_human, edition, warnings)?;
             validate_crate_types(target, target_kind_human, edition, warnings)?;
+
+            if let Some(PathValue(path)) = &target.path {
+                target.path = Some(PathValue(paths::normalize_path(path).into()));
```

Comment:
nit: Not thrilled about having to duplicate these between the regular and normalized branches. Likely not too bad at this point and not worth further churn on this PR.